src/app/pages/DashboardPage/TransactionsChartCard.tsx

```diff
@@ -59,4 +50,41 @@ const TransactionsChartCardCmp: FC<TransactionsChartCardProps> = ({ chartDuratio
   )
 }
 
-export const TransactionsChartCard = memo(TransactionsChartCardCmp)
+const TransactionsChart = memo(TransactionsChartCmp)
+
+interface TransactionsChartWrapperProps {
+  chartDuration: ChartDuration
+}
+
+export const TransactionsChartCard: FC<TransactionsChartWrapperProps> = ({ chartDuration }) => {
+  const statsParams = durationToQueryParams[chartDuration]
+  const { data } = useGetLayerStatsTxVolume(Layer.emerald, statsParams)
+  const prevData = usePrevious(data)
+
+  // React Query re-renders on isLoading change
```

Comment:
There's a flag in query `keepPreviousData`

Reply:
Right, so enabling `keepPreviousData` still reproduces the issue I am trying to solve here:

https://www.loom.com/share/e4b99f159c9f4f2ca85fb08daf8f6f65

Where graph line would just immediately switch(Today, This week), without the proper animation.

Reply:
Ahh Ok. I would still kill usePrevious, useMemo, distinctData anyway and I would just align staleTime to animation duration.

Reply:
staleTime!

Reply:
Ok, so setting `staleTime` to LineChart's `Line` `animationDuration`, still not working as expected.

https://www.loom.com/share/73f163d6a41d4f929caaa6061657840e

Reply:
The issue is `isLoading` prop, that causes re-render.

Reply:
https://www.loom.com/share/f451b34a6820442d9a99dc529da0f83a

Reply:
Got it. Resolved.